File: android/app/src/main/java/com/vxin/app/data/model/Collection.kt
package com.vxin.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class CollectionExtra(
    val file_url: String = "",
    val source_msg_id: String = "",
)

/** 收藏项（GET /api/users/me/collections） */
@Serializable
data class Collection(
    val id: String,
    val type: String = "text",      // text | image | file | video
    val content: String = "",
    val extra: CollectionExtra = CollectionExtra(),
    val created_at: Long = 0,
)

/** 收藏搜索分页响应（GET /api/users/me/collections/search） */
@Serializable
data class CollectionPage(
    val items: List<Collection> = emptyList(),
    val total: Int = 0,
    val hasMore: Boolean = false,
)

File: ios/Vxin/Data/Models/Collection.swift
import Foundation

struct CollectionExtra: Decodable {
    var fileUrl: String = ""
    var sourceMsgId: String = ""
    enum CodingKeys: String, CodingKey { case fileUrl = "file_url"; case sourceMsgId = "source_msg_id" }
    init() {}
    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        fileUrl = (try? c.decode(String.self, forKey: .fileUrl)) ?? ""
        sourceMsgId = (try? c.decode(String.self, forKey: .sourceMsgId)) ?? ""
    }
}

/// 收藏项（GET /api/users/me/collections）
struct Collection: Decodable, Identifiable {
    let id: String
    var type: String = "text"
    var content: String = ""
    var extra: CollectionExtra = CollectionExtra()
    var createdAt: Double = 0

    enum CodingKeys: String, CodingKey {
        case id, type, content, extra
        case createdAt = "created_at"
    }
    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(String.self, forKey: .id)
        type = (try? c.decode(String.self, forKey: .type)) ?? "text"
        content = (try? c.decode(String.self, forKey: .content)) ?? ""
        extra = (try? c.decode(CollectionExtra.self, forKey: .extra)) ?? CollectionExtra()
        createdAt = (try? c.decode(Double.self, forKey: .createdAt)) ?? 0
    }
}
